app/controllers/member_conversion_controller.rb

```diff
@@ -0,0 +1,202 @@
+class MemberConversionController < ApplicationController
+  include CurationConcerns::Lockable
+
+  #Promote a FileSet to a child GenericWork.
+  def to_child_work
+    begin
+      parent_work = GenericWork.find(params['parentid'])
+      file_set = FileSet.find(params['filesetid'])
+    rescue
+      flash[:notice] = "Sorry. This item no longer exists."
+      redirect_to "/works/#{parent_work.id}"
+      return
+    end
+
+    if !validate_for_switch_to_child_work(parent_work, file_set)
+      flash[:notice] = "\"#{file_set.title.first}\" can't be promoted to a child work."
+      redirect_to "/concern/parent/#{parent_work.id}/file_sets/#{file_set.id}"
+      return
+    end
+    place_in_order = parent_work.ordered_members.to_a.find_index(file_set)
+    transfer_thumbnail = (parent_work.thumbnail == file_set)
+    transfer_representative = (parent_work.representative == file_set)
+    new_child_work = GenericWork.new(title: file_set.title)
+    acquire_lock_for(parent_work.id) do
```

Comment:
I wonder if we can wrap the `acquire_lock_for` around _less_ stuff?  

As this stuff is so slow to happen... for the entire extent of the acquire_lock_for, nobody else will be able to edit the parent, and in fact their browsers (and the rails worker thread) will just hang waiting for this one to finish. 

acquire_lock_for only really needs to be around any mutations to the work. So I think everything ot line 29 (just creating the child) doesn't need to be in it. Neither does `set_thumbnail_and_rep(new_child_work, file_set, true, true)`, or `new_child_work.save!`.  Just things that might touch the parent. 

Which granted can be confusing to know what things will end up tryign to save the parent (leaky abstractions). 

I might not be totally right, please think it through and/or investigate if needed rather than just follow my directions... but I think it might pay off to try to minimize what's inside the `acquire_lock_for`, as longly-held locks are such a mess for everything.